medallion/config.py

```diff
@@ -0,0 +1,82 @@
+import json
+import pathlib
+
+import attr
+import environ
+import jsonmerge
+
+from . import backends
+
+DEFAULT_CONFFILE = "/etc/medallion.conf"
+DEFAULT_CONFDIR = "/etc/medallion.d/"
+
+
+@environ.config(prefix="TAXII")
+class TAXIIConfig(object):
+    max_page_size = environ.var(None)
+
+
+@environ.config(prefix="MEDALLION")
+class MedallionConfig(object):
+    backend = environ.group(backends.BackendConfig)
+    taxii = environ.group(TAXIIConfig)
+
+    @classmethod
+    def __strip(cls, dict_):
+        for k, v in tuple(dict_.items()):
+            if isinstance(v, dict):
+                cls.__strip(v)
+            if v is None or v == {}:
+                del dict_[k]
+
+    def as_dict(self):
+        dictified = attr.asdict(self)
+        self.__strip(dictified)
+        return dictified
+
+
+def load_config(conf_file=DEFAULT_CONFFILE, conf_dir=DEFAULT_CONFDIR):
+    # Load configuration from the default config file or a value passed in
+    if conf_file is not None:
+        conf_file_p = pathlib.Path(conf_file)
+        try:
+            config_data = json.load(conf_file_p.open())
```

Comment:
I also need to handle JSON decoding errors and raise something like a `TypeError` if the data we get back is not a JSON object (e.g. array, raw value) since json merge will complain about merging objects and other types and the config we get from `environ` is always going to be an object (ie. `dict`).